### PR TITLE
[workspace] Deprecate the tinyobjloader external

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
We are applying custom patches to our copy of tinyobyloader. As such, it needs to be badged as "internal".

Towards #17231.

- - -
This change is [Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/19880)

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
+@SeanCurtis-TRI for feature review, please.

**Review by SeanCurtis-TRI — COMMENTED** [review]
:LGTM:

> Reviewed 15 of 15 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/19880#-:-NaiIp5YBhVzart-RXM4:bmxilld)* status: 1 unresolved discussion, needs at least two assigned reviewers

___
*[`tools/workspace/tinyobjloader_internal/patches/default_texture_color.patch` line 3 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/19880#-NaiIUlW7mhLBmeU2JZv:-NaiIUlW7mhLBmeU2JZw:b-oy4ss4) ([raw file](https://github.com/RobotLocomotion/drake/blob/a110ba45d3243a553bb815bf4db4aa053342decc/tools/workspace/tinyobjloader_internal/patches/default_texture_color.patch#L3)):*
> ```Diff
> [tinyobjloader] Fix map_Kd default
> 
> If only a diffuse texture is given (map_Kd) tinyobj modulates it to 60%
> ```

BTW Given this documentation, would it make sense to remove the comment elaboration in the patch? I can imagine it's just not worth it. But right now, it is redundant.

**Review by jwnimmer-tri — COMMENTED** [review]
+@rpoyner-tri for platform review per schedule (Tuesday), please.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/19880#-:-Naj2zFFEdS3FtPM1ku7:b1vil07)* status: LGTM missing from assignee rpoyner-tri(platform)

___
*[`tools/workspace/tinyobjloader_internal/patches/default_texture_color.patch` line 3 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/19880#-NaiIUlW7mhLBmeU2JZv:-Naj2FO36m3KgaGZoCXI:b-eiuiud) ([raw file](https://github.com/RobotLocomotion/drake/blob/a110ba45d3243a553bb815bf4db4aa053342decc/tools/workspace/tinyobjloader_internal/patches/default_texture_color.patch#L3)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW Given this documentation, would it make sense to remove the comment elaboration in the patch? I can imagine it's just not worth it. But right now, it is redundant.
</blockquote></details>

Just like with real commits in `git` versus the code those commits are modifying, different people read the metadata versus the code at different times.

Once this patch is applied and someone is debugging the tinyobjloader source gdb, they will only have access to the code, not the patch description.

Likewise, it's nice to summarize the patches up top, so people working on the repository rule code don't need to grok the guts of the library code to understand the big picture about what kinds of things we are changing.

In short: I don't think they are redundant.  (I'd also like us to try to get this patch upstream.)

**Comment by jwnimmer-tri** [comment]
(I suppose we could also do "single reviewer", if you like.)

**Review by SeanCurtis-TRI — COMMENTED** [review]
I'm game either way.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/19880#-:-NajhZXd8rg5zWG_vHTH:b-3cnfwe)* status: LGTM missing from assignee rpoyner-tri(platform)

___
*[`tools/workspace/tinyobjloader_internal/patches/default_texture_color.patch` line 3 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/19880#-NaiIUlW7mhLBmeU2JZv:-Najha_J37lckLS05L72:bqbqx8h) ([raw file](https://github.com/RobotLocomotion/drake/blob/a110ba45d3243a553bb815bf4db4aa053342decc/tools/workspace/tinyobjloader_internal/patches/default_texture_color.patch#L3)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Just like with real commits in `git` versus the code those commits are modifying, different people read the metadata versus the code at different times.

Once this patch is applied and someone is debugging the tinyobjloader source gdb, they will only have access to the code, not the patch description.

Likewise, it's nice to summarize the patches up top, so people working on the repository rule code don't need to grok the guts of the library code to understand the big picture about what kinds of things we are changing.

In short: I don't think they are redundant.  (I'd also like us to try to get this patch upstream.)
</blockquote></details>

I can buy that.
